### ISSUE TITLE
WebAIM Feedback - SC 2.5.3 – Touch with Assistive Technology

### ISSUE THREAD

**Issue by jared-w-smith** [w3c/wcag21#263]
This requires that authors not introduce functionality that conflicts with screen reader overrides of touch events (or that they provide an alternative mechanism, such as “a mobile menu button”). The primary difficulty is that there is a myriad of current and yet-to-be-defined screen reader touch events that would be impossible for an author to fully test or consider. This SC therefore seems to essentially require that touch events not be used or that alternative mechanisms always be provided. We don’t believe this is the intent.

It seems that this is primarily a user agent issue – they should allow users the ability to specify whether touch events are handled by the assistive technology or are passed to the platform (as is the case with key events in Windows screen readers).

It’s likely we are misinterpreting an aspect of this SC, but it does not seem viable for inclusion in WCAG.

**Comment by DavidMacDonald** [comment]
@jared-w-smith let's talk about this, it's a real problem on the web for blind VoiceOver users. Maybe there is a way to tighten it up or make it clearer.

**Comment by jared-w-smith** [comment]
Yes, it's certainly a problem for users. I just think it's a problem that is best addressed by Apple via one feature change rather than by every mobile content author.

**Comment by DavidMacDonald** [comment]
@jared-w-smith which feature change from Apple would fix it?

**Comment by mraccess77** [comment]
@jared-w-smith another example of this might be a slider or other ARIA control that that is touch accessible and perhaps keyboard accessible.  Requiring a screen reader user to bring a physical keyboard to access the keyboard interface support is impractical.  On-screen keyboard don't exist to send these commands to Safari on iOS and even if they did Safari seems to ignore keystrokes send from physical keyboards to the web content and they are not detectable by the mobile content.  This means many ARIA controls aren't accessible -- things like combo boxes, sliders, and menus that require key presses.

**Comment by jared-w-smith** [comment]
@mraccess77 Thank you for the additional details.

Combo boxes, sliders, and menus are standard, common controls and have full, standards-based accessibility support - except in these particular screen readers. I would hope that before we go down the path of asking all authors to provide alternatives for standard controls that significant efforts be made with the mobile screen reader vendors to see if they can address this at the user agent level.

**Comment by patrickhlauke** [comment]
@jared-w-smith 
> they should allow users the ability to specify whether touch events are handled by the assistive technology or are passed to the platform (as is the case with key events in Windows screen readers).

the complication here is that with a keyboard, you have quite a big set of keys that still allow you to provide some form of explicit switching (to turn the AT's control scheme back on). with touch, you're pretty much limited to "the user made a gesture" on the one single input interface. if you said the AT needs to provide some switching mechanism, you're still going to have at least one specific gesture (to turn AT control back on) that you'd need to tell authors not to use, and THIS may be different across different UAs/ATs.

> This SC therefore seems to essentially require that touch events not be used or that alternative mechanisms always be provided. We don’t believe this is the intent.

not that touch events not be used, but that another way of interacting needs to be present (same way that 2.1.1 Keyboard doesn't mean "don't use mouse events"). an alternative would need to be provided anyway assuming the web page wants to pass existing 2.1.1 Keyboard. and for the most part, following 2.1.1 (and NOT making naive "oh, this browser has touch, so i can ignore mouse/keyboard, since there are no scenarios where a touchscreen is present in combination with any other input mechanism" assumptions, which are incorrect anyway - phones/tablets with paired keyboard/mouse, touch-enabled laptops/desktops, etc) will already pass this, with one crucial difference: currently you can pass 2.1.1 by listening to arbitrary key events. On touchscreen devices without paired external keyboard, this is not possible (and even WITH a paired keyboard, in some cases - iOS/Safari/VO for instance - key events aren't triggered unless the user is focused on an input that expects keyboard input).

in short (as I wrote over in https://github.com/w3c/wcag21/issues/195#issuecomment-298203325):

> - things need to be focusable
> - things can't rely on gestures other than activation (no swipes, multitouches, etc as they're always intercepted by touch AT unless user does a "passthrough" gesture, where available)
> - things can't rely on user being able to arbitrarily hit a keyboard key (this nuance differentiates it from the regular 2.1.1 Keyboard, where this *would* be acceptable to pass)

**Comment by awkawk** [comment]
@jared-w-smith Thanks for the comment - the Working Group has not reached consensus on this proposed SC so it is deferred for future consideration.
